proposals/0672-declarative-xcode-projects.md

```diff
@@ -0,0 +1,92 @@
+
+It also uses a proprietary syntax: it is not a XML, a JSON or anything standard. Preexistent tools can't really highlight changes in a meaningful way, making the Xcodeproj files just a wall of test which can feel daunting for the final user.
+
+The current proposal suggests to move away from manually managing `Xcodeproj` files in favor of a more declarative description. There are different tools in the OSS space that allows to describe the project declaratively and that then reads that description and generate a valid Xcodeproj accordingly.
+
+The suggested too is [Tuist.io](https://tuist.io): it allows to describe the project in Swift, leveraging type checking, syntax highlights and code completion. It also uses the default language for Apple development, which is familiar to all the iOS developers and it is executable code, providing more flexibility than a pure markup language.
+
+## Basic example
+
+You can have a look at it in action in this PR:
+
+- https://github.com/facebook/react-native/pull/37952
+
+The idea is to describe the project with a declarative approach using Swift. Tuist allows you to load the `Project.swift` file in an instance of Xcode throught the `tuist edit` command. This allows to leverage code completion, type checking and syntax highlight to ensure the correctness of the project.
+
+## Motivation
+
+Managing Xcodeproj files is error prone and time consuming. Those files are source of frustrations for teams using them and they have been reported as one of the main pain point when users have to migrate to a new version of React Native.
+
+Xcode generates projects using random UUIDs to identify the objects that constitutes a project and to refer to them in the project description file itself. These UUIDs are a nightmare to work with, especially when large teams use the same project's file: when two engineers adds two files in the same project, Xcode may create two different UUIDs for the same file and it is usually messy to handle that conflict.
+
+Managing Xcodeproj manually is also an old and not efficient way to manage projects: even Apple realized that and started developing Swift Package Manager (SwiftPM) to replace the project for developing libraries.
+
+Modern approaches rely on a declarative syntax to describe how the project should look like and they rely on tools that can generate a valid Xcodeproj starting from that description. For example, [XCodeGen](https://github.com/yonaskolb/XcodeGen) uses YAML to describe the project, while [Tuist.io](https://tuist.io) uses directly swift to achieve the same result.
+
+The proposed solution is, indeed, to use [Tuist.io](https://tuist.io). It is an OSS tool, with a vibrant community and backed and supported by several companies. Some of its maintainers are also React Native core contributors.
+
+## Detailed design
```

Comment:
I'm missing a high-level explanation of what "using Tuist" means for our users, between the motivation (why) and detailed design (how). What will be different about the experience? How will users need to adjust? Does this leave out any groups of users? (e.g. users who have some specific need to keep managing their own Xcodeproj)

Reply:
Thanks for the feedback! I will look for some time to expand this section.

> Does this leave out any groups of users? (e.g. users who have some specific need to keep managing their own Xcodeproj)

To address this concern, no. What you can do while managing the Xcodeproj manually can be achieved better with Tuist. Also, nobody in the industry manages the Xcodeproj manually, because of the problem highlighted above.